### ISSUE TITLE
find replacement for `buf_redux` library

### ISSUE THREAD

**Issue by marcmo** [esrlabs/chipmunk#1835]
library is not supported anymore

**Comment by AmmarAbouZor** [comment]
I think one of the best candidates is the crate [bytes](https://github.com/tokio-rs/bytes)
I'll make some research about it and try to replace it while exploring the source code

**Comment by AmmarAbouZor** [comment]
I don't think we can use the crate `bytes` to replace `buf_redux` because it  doesn't support handling reading from a file directly but rather it works with in-memory bytes only.

I've found the maintained fork [buffer_redux](https://crates.io/crates/buffer-redux), which still hasn't commited changes on the original code besides updating the dependencies, adding CI Pipelines and applying cargo clippy fixes.

I would suggest trying this crate in a draft PR where we can test it and then we can merge if everything is still working fine. We will need to to merge `buffer_redux` change in [Dmitry's fork](https://github.com/DmitryAstafyev/buf_redux.git) as well.

**Comment by DmitryAstafyev** [comment]
@AmmarAbouZor last time we discussed with @marcmo the possibility to maintain `buf_redux` crate as well. It isn't a huge lib and already includes some of our changes.
@marcmo it looks like it make sense to make a fork of `buf_redux` in `esrlabs` and re-register it also on `crates.io`. What do you think?

**Comment by AmmarAbouZor** [comment]
It would be very interesting to maintain the crate, since it's really low level and has its share of low level implementation  across different operating systems and unsafe code.

**Comment by AmmarAbouZor** [comment]
Already implemented
